### PR TITLE
Just enough VGA and BIOS to get Krondor to the not-enough-memory exit

### PR DESCRIPTION
**Pull request by JorisVanEijden**
- Introduce BIOS memory as a separate class (segment 0x40)
- Introducs VGA Bios to VgaCard
- Add 3 standard fonts to VGA BIOS (lazy loaded)
- Fix negative dos files index

This lays a foundation for further bios & vga additions

### CODE REVIEW

**Comment by maximilien-noal** [comment]
Overall, this looks very good to me. :D

The dedicated BIOS class is a very nice addition, that the previous code was calling for.

File: src/Spice86.Core/Emulator/Memory/Memory.cs [hunk 1]
```diff
@@ -176,4 +176,23 @@ public class Memory {
         CurrentlyWritingByte = value;
         _writeBreakPoints.TriggerMatchingBreakPoints(address);
     }
+
+    // Overloads to work with segment:offset addresses.
+    public uint GetUint32(ushort segment, ushort offset) => 
+        GetUint32((uint)((segment << 4) + offset));
```
Comment (kevinferrare) [inline src/Spice86.Core/Emulator/Memory/Memory.cs:182]: Use ToPhysicalAddress.ToPhysicalAddress to compute the actual physical address.

File: src/Spice86.Core/Emulator/Memory/MemoryMap.cs [hunk 2]
```diff
@@ -33,4 +24,9 @@ public static class MemoryMap {
     public const int MonochromeTextVideoMemoryLength = 32767;
 
     public const int MonochromeTextVideoMemorySegment = 0xB000;
+    
+    /// <summary>
+    /// Segment where VGA BIOS is stored.
+    /// </summary>
+    public const ushort VideoBios = 0xC000;
```
Comment (kevinferrare) [inline src/Spice86.Core/Emulator/Memory/MemoryMap.cs:31]: VideoBios => VideoBiosSegment?

**Comment by kevinferrare** [comment]
I tried the change with DUNE (CD version) and it crashes. It seems at some point Gui.Width is 0:
 ---> System.DivideByZeroException: Attempted to divide by zero.
   at ushort Spice86.Core.Emulator.InterruptHandlers.Input.Mouse.MouseInt33Handler.RestrictValue(ushort value, ushort maxValue, ushort min, ushort max) in src/Spice86.Core/Emulator/InterruptHandlers/Input/Mouse/MouseInt33Handler.cs:line 186
   at void Spice86.Core.Emulator.InterruptHandlers.Input.Mouse.MouseInt33Handler.GetMousePositionAndStatus() in src/Spice86.Core/Emulator/InterruptHandlers/Input/Mouse/MouseInt33Handler.cs:line 55
...

File: src/Spice86.Core/Emulator/Memory/Memory.cs (hunk 1 above)
Comment (JorisVanEijden) [reply src/Spice86.Core/Emulator/Memory/Memory.cs]: Oops, hadn't noticed those. Will do.

File: src/Spice86.Core/Emulator/Memory/MemoryMap.cs (hunk 2 above)
Comment (JorisVanEijden) [reply src/Spice86.Core/Emulator/Memory/MemoryMap.cs]: Good suggestion. I renamed it.

**Comment by JorisVanEijden** [comment]
> I tried the change with DUNE (CD version) and it crashes. It seems at some point Gui.Width is 0: ---> System.DivideByZeroException: Attempted to divide by zero. at ushort Spice86.Core.Emulator.InterruptHandlers.Input.Mouse.MouseInt33Handler.RestrictValue(ushort value, ushort maxValue, ushort min, ushort max) in src/Spice86.Core/Emulator/InterruptHandlers/Input/Mouse/MouseInt33Handler.cs:line 186 at void Spice86.Core.Emulator.InterruptHandlers.Input.Mouse.MouseInt33Handler.GetMousePositionAndStatus() in src/Spice86.Core/Emulator/InterruptHandlers/Input/Mouse/MouseInt33Handler.cs:line 55 ...

Interesting. So for some reason it's suddenly having DX 0 when calling SetMouseHorizontalMinMaxPosition?

What if you change the startup VideoMode from 0x03 to 0x13 in `Spice86.Core.Emulator.Memory.Bios.Bios` ?

**Comment by maximilien-noal** [comment]
> I tried the change with DUNE (CD version) and it crashes. It seems at some point Gui.Width is 0: ---> System.DivideByZeroException: Attempted to divide by zero. at ushort Spice86.Core.Emulator.InterruptHandlers.Input.Mouse.MouseInt33Handler.RestrictValue(ushort value, ushort maxValue, ushort min, ushort max) in src/Spice86.Core/Emulator/InterruptHandlers/Input/Mouse/MouseInt33Handler.cs:line 186 at void Spice86.Core.Emulator.InterruptHandlers.Input.Mouse.MouseInt33Handler.GetMousePositionAndStatus() in src/Spice86.Core/Emulator/InterruptHandlers/Input/Mouse/MouseInt33Handler.cs:line 55 ...

With commit [180261809bafea7cffae817f2d113cdfe73ff4eb](https://github.com/OpenRakis/Spice86/pull/248/commits/180261809bafea7cffae817f2d113cdfe73ff4eb), I didn't have any problem with DUNE CD v3.7.
Tested witthout any sound, with MIDI, and finally MIDI + PCM,
Full intro, gameplay, loaded a save, ordered Fremens around -> nothing to report.

**Comment by maximilien-noal** [comment]
Risk avoided on master with the latest commit.

**Pull request merged**